class-00/lab-a/lab/js/app.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+const names = ['bag', 'banana', 'bathroom', 'boots', 'breakfast', 'bubblegum', 'chair', 'cthulhu', 'dog-duck', 'dragon', 'pen', 'pet-sweep', 'scissors', 'shark', 'sweep', 'tauntaun', 'unicorn', 'usb', 'water-can', 'wine-glass'];
+
+const leftImage = document.getElementById('left');
+const centerImage = document.getElementById('center');
+const rightImage = document.getElementById('right');
+
+let allProducts = [];
+const container = document.getElementById('image_container');
+const viewed = [];
+const labels = [];
+const pics = [leftImage, centerImage, rightImage];
+const list = document.getElementById('productlist');
+let totalClicks = 0;
+const views = [];
+const votes = [];
+
+function Product(name) {
+  this.name = name;
+  this.path = 'img/' + name + '.jpg';
+  this.votes = 0;
+  this.views = 0;
+  allProducts.push(this);
+}
+
+function makeRandom() {
+  return Math.floor(Math.random() * names.length);
+}
+
+function displayPics(){
+  while(viewed.length < 6){
+    const rando = makeRandom();
+    // console.log('trying it here',rando);
+    while(!viewed.includes(rando)){
+      viewed.push(rando);
+      // console.log('trying it here',rando);
+    }
+    // console.log('trying it here',rando);
+  }
+  // console.log(rando);
+  // TODO: In a sentence or two, explain why the previous line of code threw an error when we changed the variable declaration from `var to `const`.
+  // It looks like with let rando = makeRandom() is scoped within the while loop only
+  console.log(viewed);
+
+  for (let i = 0; i < 3; i++){
+    const temp = viewed.shift();
+    pics[i].src = allProducts[temp].path;
+    pics[i].id = allProducts[temp].name;
+    allProducts[temp].views += 1;
+  }
+}
+
+function handleClick(event) {
+  if (event.target.id === 'image_container') {
+    return alert('Be sure to click directly on an image!!');
+  }
+  totalClicks += 1;
+  if(totalClicks > 24) {
+    container.removeEventListener('click', handleClick);
+    container.style.display = 'none';
+    showList();
+    makeChart();
+  }
+  for(let i = 0; i < names.length; i++){
+    if(event.target.id === allProducts[i].name) {
+      allProducts[i].votes += 1;
+      console.log(`${event.target.id} has ${allProducts[i].votes} votes in ${allProducts[i].views} views`);
+    }
+  }
+  localStorage.busmall = JSON.stringify(allProducts);
+  localStorage.busmallProducts = JSON.stringify(allProducts);
+  displayPics();
+}
+
+function showList() {
+  for(let i = 0; i < allProducts.length; i++) {

```

Comment:
const is only accessible within the code block it was declared in. For example you would get undefined if you tried to console.log after the closing curly brace on line 81.